tests/tests_integration/test_api/test_data_modeling/test_instances.py

```diff
@@ -530,6 +556,66 @@ def test_query_oscar_winning_actors_before_2000(
         assert all(actor.properties.get(actor_id, {}).get("wonOscar") for actor in result["actors"])
         assert result["actors"] == sorted(result["actors"], key=lambda x: x.external_id)
 
+    def test_apply_retrieve_list_search_aggregate_in_units(
+        self, cognite_client: CogniteClient, integration_test_space: Space, unit_view: View
+    ) -> None:
+        node = NodeApply(
+            space=integration_test_space.space,
+            external_id="nodewrite_units_test",
+            sources=[
+                NodeOrEdgeData(
+                    unit_view.as_id(),
+                    {"pressure": 1.1},
+                )
+            ],
+        )
+
+        source = SourceDef.from_view_id(
+            unit_view.as_id(), [PropertyUnitReference("pressure", UnitReference("pressure:pa"))]
+        )
+
+        created = cognite_client.data_modeling.instances.apply(node, replace=True)
+
+        retrieved = cognite_client.data_modeling.instances.retrieve(created.nodes.as_ids(), sources=[source])
+        assert retrieved.nodes
+        assert abs(retrieved.nodes[0]["pressure"] - 1.1 * 1e5) < 1e-5
+
+        is_node = filters.Prefix(["node", "externalId"], node.external_id)
```

Comment:
Just in case someone (me) would add another node with `node.external_id + "-troll"`
```suggestion
        is_node = filters.Equals(["node", "externalId"], node.external_id)
```